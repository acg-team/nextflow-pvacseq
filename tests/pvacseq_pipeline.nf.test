nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"

    test("Full pipeline test with online VEP and IEDB directory check") {
        when {
            params {
                outdir = "$projectDir/tests/pipeline_outdir"
                fasta = "$projectDir/tests/data/chrY.fa"
                hla_csv = "$projectDir/tests/data/small_hla.csv"
                input = "$projectDir/tests/data/input/"
                pvacseq_iedb = "$projectDir/iedb" // download iedb to this folder
                extra_vep_args = "--database" // Use online VEP annotation
                vep_genome = "GRCh38"
                vep_cache_version = "102"
                vep_species = "homo_sapiens"
                pvacseq_algorithm = "all"
            }
        }
        then {
            def iedb_dir_count = 0
            path("${params.pvacseq_iedb}").eachFileRecurse(groovy.io.FileType.FILES) {
                iedb_dir_count++
            }
            def outdir_count = 0
            path("${params.outdir}").eachFileRecurse(groovy.io.FileType.FILES) {
                outdir_count++
            }
            def mhc_i_filtered = "$params.outdir/pvactools/TCGA-F5-6814-01A-31D-1924-10/MHC_Class_I/TCGA-F5-6814-01A-31D-1924-10.filtered.tsv"
            assert workflow.success
            assert file(mhc_i_filtered).exists()

            def mhc_i_lines = path(mhc_i_filtered).readLines()
            assert mhc_i_lines.size() == 2 : "MHC I filtered file should have 1 data lines (header + 1 data)"
            assert snapshot(
                // Number of successful tasks
                workflow.trace.succeeded().size(),
                // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                removeNextflowVersion("$params.outdir/pipeline_info/nf_core_pipeline_software_versions.yml"),
                // For IEDB we only count amount of files
                iedb_dir_count,
                // Most of the files are unstable
                outdir_count
            ).match()
        }
    }
}
