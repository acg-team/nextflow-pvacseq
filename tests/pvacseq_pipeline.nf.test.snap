{
    "Full pipeline test with online VEP and IEDB directory check": {
        "content": [
            10,
            {
                "ENSEMBLVEP_VEP": {
                    "ensemblvep": 113.0,
                    "tabix": 1.21
                },
                "MAF2VCF": {
                    "maf2vcf": "1.6.21"
                },
                "PVACTOOLS_PVACSEQ": {
                    "pvactools": "5.3.1"
                },
                "Workflow": {
                    "nextflow_pvacseq": "v1.0dev"
                }
            },
            207427,
            207661
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-09-02T12:52:19.750386944"
    }
}